modules/build/src/main/scala/scala/build/ConsoleBloopBuildClient.scala

```diff
@@ -46,14 +47,14 @@ class ConsoleBloopBuildClient(
   ): Option[(Either[String, os.Path], bsp4j.Diagnostic)] =
     diagnosticMappings.get(path).collect {
       case (originalPath, lineOffset)
-          if diag.getRange.getStart.getLine + lineOffset >= 0 &&
-          diag.getRange.getEnd.getLine + lineOffset >= 0 =>
+          if scalaLineToScLine(lineOffset)(diag.getRange.getStart.getLine).isDefined &&
+          scalaLineToScLine(lineOffset)(diag.getRange.getEnd.getLine).isDefined =>
         val start = new bsp4j.Position(
-          diag.getRange.getStart.getLine + lineOffset,
+          scalaLineToScLine(lineOffset)(diag.getRange.getStart.getLine).get,
           diag.getRange.getStart.getCharacter
         )
         val end = new bsp4j.Position(
-          diag.getRange.getEnd.getLine + lineOffset,
+          scalaLineToScLine(lineOffset)(diag.getRange.getEnd.getLine).get,
           diag.getRange.getEnd.getCharacter
         )
         val range = new bsp4j.Range(start, end)
```

Comment:
...or you could avoid the whole `.isDefined` => `.get` with something like:
```suggestion
    diagnosticMappings.get(path)
      .map { case (originalPath, lineOffset) =>
        (
          originalPath,
          scalaLineToScLine(lineOffset)(diag.getRange.getStart.getLine),
          scalaLineToScLine(lineOffset)(diag.getRange.getEnd.getLine)
        )
      }
      .collect {
        case (originalPath, Some(scLineStart), Some(scLineEnd)) =>
          val start       = new bsp4j.Position(scLineStart, diag.getRange.getStart.getCharacter)
          val end         = new bsp4j.Position(scLineEnd, diag.getRange.getEnd.getCharacter)
          val range       = new bsp4j.Range(start, end)
```